### ISSUE TITLE
`new-cap` reporting wrong position for lowercase constructors

### ISSUE THREAD

**Issue by radiovisual** [eslint/eslint#5514]
I seem to be getting some sort of false positive on `new-cap`. ESLint is complaining about a constructor starting with a lowercase letter at position `6:50`:
```
6:50  error  A constructor name should not start with a lowercase letter  new-cap
```

...but that position does not have any constructors with lowercase letters:

```
5 |
6 |		editor.addCommand('mediawrapper', new CKEDITOR.dialogCommand('mediawrapperDialog'));
7 |                                              |
8 |                                              |
9 |                                              Position 6:50 --> no lowercase letters
```

To see the full source code, see the answers below:

**What version of ESLint are you using?**
2.3.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
- configuration: [eslint-config-xo](https://github.com/sindresorhus/eslint-config-xo/blob/master/index.js)
- parser: [XO](https://github.com/sindresorhus/xo)

**What did you do? Please include the actual source code causing the issue.**

I ran ESLint (via XO) on the following code:

```js
var CKEDITOR = {};

CKEDITOR.plugins.add('mediawrapper', {
	icons: 'mediawrapper',
	init: function (editor) {
		editor.addCommand('mediawrapper', new CKEDITOR.dialogCommand('mediawrapperDialog'));
	}
});
```
**What did you expect to happen?**

I expected not to see an error on line 6, because there are no constructors starting with a lowercase letter in my code.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  6:50  error  A constructor name should not start with a lowercase letter  new-cap
```

**Comment by ilyavolodin** [comment]
Hmm.. interesting. Looks like positions are reported incorrectly and also looks like rule is having an issue. XO's configuration for `new-cap` is `'new-cap': [2, {newIsCap: true, capIsNew: true}]`

**Comment by michaelficarra** [comment]
The `d` is lowercase. The rule is working as expected.

**Comment by ilyavolodin** [comment]
Oh I see. It's checking for the property name as well as object name. Good point. Although it looks like location is still reported incorrectly.

**Comment by radiovisual** [comment]
I see that in Issue #3172 there was some way to disable `new-cap` on properties, but I don't find any documentation on this in order to test it in cases like this. There is still the issue of the incorrect location, so if you want, I can change the name of this issue to focus on the incorrect placement.

**Comment by michaelficarra** [comment]
@ilyavolodin I just tried it in the online demo, and the position is reported correctly.

**Comment by radiovisual** [comment]
Yeah, I just ran it in the online demo and it came back with position `6:56`, which is the correct position for the lowercase `d`.

**Comment by kaicataldo** [comment]
@radiovisual Were you able to figure out what was going on?

**Comment by radiovisual** [comment]
@kaicataldo , no. It was determined that the `new-cap` rule was working correctly (since it was also looking at the lowercase `d` on the property), and that my version of ESLint was reporting the wrong location of the lowercase `d`. 

So really, this issue should be renamed to: **`new-cap` reporting wrong location for lowercase constructors.**

**Comment by kaicataldo** [comment]
@radiovisual  Gotcha - are you still able to reproduce it with the latest version of ESLint?

**Comment by radiovisual** [comment]
@kaicataldo , yes, I just ran ESLint `v2.8.0`, on the same exact code sample, and I get the exact same output that I did with `v2.3.0`. The position is still off:

<pre>
6:50  error  A constructor name should not start with a lowercase letter  new-cap
</pre>

```
5 |
6 |		editor.addCommand('mediawrapper', new CKEDITOR.dialogCommand('mediawrapperDialog'));
7 |                                              |
8 |                                              |
9 |                                              Position 6:50 --> no lowercase letters
```

**Comment by kaicataldo** [comment]
@radiovisual Alright, thanks for the info and for following up :+1:

**Comment by pmcelhaney** [comment]
Tried with ESLint 2.8.0, ESLint 3.1.1, and XO 0.16.0 and couldn't reproduce. It always reports the error at character 56. 

```sh
$ ./node_modules/eslint/bin/eslint.js ckeditor.js 

/Users/patrickmcelhaney/code/new-cap-test/ckeditor.js
  6:56  error  A constructor name should not start with a lowercase letter  new-cap
  8:21  error  A constructor name should not start with a lowercase letter  new-cap

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by nzakas** [comment]
It appears this has been solved, so closing.
